app/deliveryoptions/delivery_options.py

```diff
@@ -0,0 +1,432 @@
+        record_data: The record object
+        api_surrogate_data: List of surrogate data
+
+    Returns:
+        str: The processed heading HTML
+    """
+    return html_builder(
+        delivery_option_data, record_data, api_surrogate_data=api_surrogate_data
+    )
+
+
+def surrogate_link_builder(surrogates: List) -> List[Any]:
+    """
+    Extract surrogate links and AV media links from surrogate data.
+
+    Args:
+        surrogates: The list of surrogate data
+
+    Returns:
+        List[Any]: A list containing surrogate links (including AV media links)
+    """
+    surrogate_list = []
+
+    for s in surrogates:
+        if s["xReferenceURL"]:
+            surrogate_list.append(s["xReferenceURL"])
+
+    return surrogate_list
+
+
+def construct_delivery_options(
+    api_result: List, record: Record, request: HttpRequest
+) -> Dict[str, Any]:
+    """
+    Construct delivery options based on record and request information.
+
+    This is the main function called from records.py to build the delivery options
+    for a record.
+
+    Args:
+        api_result: List of delivery options
+        record: The record object
+        request: The HTTP request
+
+    Returns:
+        Dict[str, Any]: The constructed delivery options
+    """
+
+    if api_length := len(api_result) > 1:
+        raise ValueError(
+            f"Too many results ({api_length}) from DORIS database for IAID {record.iaid}"
+        )
+
+    delivery_options_context_dict = {}
+
+    reader_type = get_reader_type(request)
+
+    delivery_options_context_dict["reader_type"] = reader_type
+
+    do_dict = read_delivery_options(settings.DELIVERY_OPTIONS_CONFIG)
+
+    # Surrogate links is always present as a list, which can be empty
+    do_surrogate = surrogate_link_builder(api_result[0]["surrogateLinks"])
+
+    if (
+        api_result[0]["options"]
+        == AvailabilityCondition.ClosedRetainedDeptKnown
+    ):
+        # Special case. Sometimes, for record type 14 (ClosedRetainedDeptKnown), the department name does not match
+        # any entry in the DEPARTMENT_DETAILS dictionary. This shouldn't happen but it does. Therefore, reset the type
+        # with that for AvailabilityCondition.ClosedRetainedDeptUnKnown
+        if not get_dept(record.reference_number, "deptname"):
+            api_result[0][
+                "options"
+            ] = AvailabilityCondition.ClosedRetainedDeptUnKnown
+
+    # Get the specific delivery option for this artefact
+    delivery_option = get_record(do_dict, api_result[0]["options"])
+
+    reader_option = delivery_option["readertype"][reader_type]
+
+    if heading := reader_option.get("heading"):
+        delivery_options_context_dict["do_heading"] = heading_builder(
+            heading, record, do_surrogate
+        )
+
+    if text := reader_option.get("description"):
+        delivery_options_context_dict["do_description"] = description_builder(
+            text, record, do_surrogate
+        )
```

Comment:
```suggestion
    if description := reader_option.get("description"):
        delivery_options_context_dict["do_description"] = description_builder(
            description, record, do_surrogate
        )
```
OR

```suggestion
    if value := reader_option.get("description"):
        delivery_options_context_dict["do_description"] = description_builder(
            value, record, do_surrogate
        )
```
Becoming difficult to read text(var) <--> description(attribute) . Wouldn't it be nicer to have same name as that of the attribute or a generic value.

Reply:
All changed now - see V4